### PR TITLE
Reprovisioning support for dps

### PR DESCRIPTION
**Pull request by timtay-microsoft**
This PR includes all feature implementation needed to enable reprovisioning.

This includes all e2e tests.

### CODE REVIEW

**Comment by timtay-microsoft** [comment]
Temporarily disabling non-provisioning tests. DO NOT MERGE YET

**Comment by timtay-microsoft** [comment]
//AzurePipelines run all

**Comment by timtay-microsoft** [comment]
//AzurePipelines run

File: e2e/test/netstandard20/ProvisioningE2ETests.cs [hunk 1]
```diff
@@ -50,274 +47,664 @@ public enum EnrollmentType
             Group
         }
 
+        public enum TransportProtocol
```
Comment (jenoola) [inline e2e/test/netstandard20/ProvisioningE2ETests.cs]: Why are you introducing this enum instead of using existing one?

Comment (timtay-microsoft) [reply e2e/test/netstandard20/ProvisioningE2ETests.cs]: I'll see if I can rewrite these tests to reuse that enum, sure

**Pull request merged**